components/raftstore-v2/src/worker/refresh_config.rs

```diff
@@ -117,13 +133,88 @@ where
             "to" => self.raft_pool.state.expected_pool_size
         );
     }
+
+    fn resize_apply_pool(&mut self, size: usize) {
+        let current_pool_size = self.apply_pool.get_pool_size();
+        if current_pool_size == size {
+            return;
+        }
+
+        // It may not take effect immediately. See comments of
+        // ThreadPool::scale_workers.
+        // Also, the size will be clamped between min_thread_count and the max_pool_size
+        // set when the apply_pool is initialized. This is fine as max_pool_size
+        // is relatively a large value and there's no use case to set a value
+        // larger than that.
+        self.apply_pool.scale_pool_size(size);
+        let (min_thread_count, max_thread_count) = self.apply_pool.thread_count_limit();
+        if size > max_thread_count || size < min_thread_count {
+            warn!(
+                self.logger,
+                "apply pool scale size is out of bound, and the size is clamped";
+                "size" => size,
+                "min_thread_limit" => min_thread_count,
+                "max_thread_count" => max_thread_count,
+            );
```

Comment:
Should we check the size before actually scale_pool_size is called.  If the size is out of count limit, we just log error and fail the operation.

Reply:
When code reaches here, the actual toml file has been changed.
I think clamp  is fine as there's no use case to exceed the max thread count which is a large value (max of 4 and cpu core numbers).